hub-saml/src/main/java/uk/gov/ida/saml/hub/api/HubTransformersFactory.java

```diff
@@ -434,7 +448,7 @@ public DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer getDecoratedSaml
             IdExpirationCache<String> assertionIdCache,
             String hubEntityId) {
         IdpResponseValidator validator = new IdpResponseValidator(this.getSamlResponseSignatureValidator(idpSignatureValidator),
-            this.getSamlResponseAssertionDecrypter(keyStore),
+            new FederationAssertionsDecryptor(this.getSamlResponseAssertionDecrypters(keyStore), new IdpFailedSamlAssertionsDecryption()),
```

Comment:
The indenting on these arguments make it really hard to see what's happening. GitHub won't let me make such a big suggestion but it might be easier to digest if it looked more like:
```
        IdpResponseValidator validator = new IdpResponseValidator(
                this.getSamlResponseSignatureValidator(idpSignatureValidator),
                new FederationAssertionsDecryptor(
                        this.getSamlResponseAssertionDecrypters(keyStore),
                        new IdpFailedSamlAssertionsDecryption()
                ),
                getSamlAssertionsSignatureValidator(idpSignatureValidator),
                new EncryptedResponseFromIdpValidator<>(new SamlStatusToIdaStatusCodeMapper()),
                new DestinationValidator(expectedDestinationHost, expectedEndpoint),
                getResponseAssertionsFromIdpValidator(assertionIdCache, hubEntityId)
        );

        return new DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer(
            validator,
            new IdaResponseFromIdpUnmarshaller(
                    new IdpIdaStatusUnmarshaller(),
                    getAssertionToPassthroughAssertionTransformer()
            )
        );
```